docs/src/index.md

```diff
@@ -1,72 +1,96 @@
 
-**ChainRules is an AD-independent collection of rules to use in a differentiation system.**
+## ChainRules ecosystem organisation
 
+The ChainRules ecosystem comprises:
+- [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl): a system for defining rules, and a collection of tangent types.
+- [ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl): a collection of rules for Julia Base and standard libraries.
+- [ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl): utilities for testing rules using finite differences.
 
-!!! note "The whole field is a mess for terminology"
-    It isn't just ChainRules, it is everyone.
-    Internally ChainRules tries to be consistent.
-    Help with that is always welcomed.
+AD systems depend on ChainRulesCore.jl to get access to tangent types and the core rule definition functionality (`frule` and `rrule`), and on ChainRules.jl to benefit from the collection of rules for Julia Base and the standard libraries.
 
-!!! terminology "Primal"
-    Often we will talk about something as _primal_.
-    That means it is related to the original problem, not its derivative.
-    For example in `y = foo(x)`, `foo` is the _primal_ function, and computing `foo(x)` is doing the _primal_ computation.
-    `y` is the _primal_ return, and `x` is a _primal_ argument.
-    `typeof(y)` and `typeof(x)` are both _primal_ types.
+Packages that just want to define rules only need to depend on [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl), which is an exceptionally light dependency.
+They should also have a test-only dependency on [ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl) to test the rules using finite differences.
 
+Note that the packages with rules do not have to depend on AD systems, and neither do the AD systems have to depend on individual packages.
 
-## `frule` and `rrule`
+## Key functionality
+
+Consider a relationship $y = f(x)$, where $f$ is some function.
+Computing $y$ from $x$ is the original problem, called the _primal_ computation, in contrast to the problem of computing derivatives.
+We say that the _primal function_ $f$ takes a _primal input_ $x$ and returns the _primal output_ $y$.
+
+ChainRules rules are concerned with propagating _tangents_ of primal inputs to _tangents_ of primal outputs (`frule`, from forwards mode AD), and propagating _cotangents_ of primal outputs to _cotangents_ of primal inputs (`rrule`, from reverse mode AD).
+To be able to do that, ChainRules also defines a small number of tangent types to represent tangents and cotangents.
+
+!!! note "Tangents and cotangents"
+    Strictly speaking tangents, $ẋ = \frac{dx}{da}$, are propagated in `frule`s, and cotangents, $x̄ = \frac{da}{dx}$, are propagated in `rrule`s.
+    However, we do not distinguish between the two in practice: both are represented by the same tangent types, and we sometimes (often?) refer to cotangents as tangents.
```

Comment:
```suggestion
    However, in practice there is rarely a need to distinguish the two: both are represented by the same tangent types.
    Thus, except when the detail might clarify, we refer to both as tangents.
```

Reply:
Tangents and cotangents need a proper, accessible introduction imo. Can explain it first as "propagating derivatives" of some "outermost function" `F` wrt the inputs of a given function `g` to the derivatives of this outermost function `F` wrt the outputs of `g` (frule) or the other way around (rrule).

Reply:
Yes, I think we should have that in the maths section. I will open an issue, it's out of scope of rewriting the intro IMO.